pybop/costs/fitting_costs.py

```diff
@@ -279,3 +280,78 @@ def evaluateS1(self, x):
             If an error occurs during the calculation of the cost or gradient.
         """
         raise NotImplementedError
+
+
+class MAP(BaseLikelihood):
+    """
+    Maximum a posteriori cost function.
+
+    Computes the maximum a posteriori cost function, which is the sum of the
+    negative log likelihood and the log prior.
+
+    Inherits all parameters and attributes from ``BaseLikelihood``.
+
+    """
+
+    def __init__(self, problem, likelihood, sigma=None):
+        super(MAP, self).__init__(problem)
+        self.sigma0 = sigma
+        if self.sigma0 is None:
+            self.sigma0 = []
+            for param in self.problem.parameters:
+                self.sigma0.append(param.prior.sigma)
+        self.likelihood = likelihood(problem=self.problem, sigma=self.sigma0)
```

Comment:
Please add a check on the `likelihood` input and a message to the user for any cost object that is not an instance of `pybop.GaussianLogLikelihoodKnownSigma`.